### PR TITLE
fix: Remove leading underscore

### PR DESCRIPTION
**Pull request by arbourd**
`go-envconfig` made a change in v0.8.x where any leading underscore (ex: `_APP_NAME`) is invalid and will break env var parsing. 

This commit removes the leading prefixed underscore from built-in options, and sets the default prefix string to `VK_`.

All new prefixes must end with an underscore when this change is released and adopted. For example, an app that used `_MY_ENV` for an env var and `PREFIX` for the prefix, but be changed to `PREFIX_` for the prefix and `MY_ENV` for the env var.

### CODE REVIEW

**Comment by arbourd** [comment]
If released, we'd wanna bump this a minor and hope that its the final compatibility release we need for vektor now that it's deprecated.

**Comment by cohix** [comment]
@arbourd in the `finalize` method, could you add a check to ensure that any provided prefix includes a trailing underscore, and add it automatically if it doesn't?

File: vk/options.go [hunk 1]
```diff
@@ -75,6 +76,11 @@ func (o *Options) ShouldUseHTTP() bool {
 
 // finalize "locks in" the options by overriding any existing options with the version from the environment, and setting the default logger if needed
 func (o *Options) finalize(prefix string) {
+	// Append trailing _ if prefix is missing one
+	if !strings.HasSuffix(prefix, "_") {
+		prefix = prefix + "_"
+	}
+
```
Comment (arbourd) [inline vk/options.go:83]: We could describe this in the docs but I think we'd want people to use the prefix `APP_` over `APP`, where this fallback exists for backwards compatibility reasons.

File: vk/test/routes_test.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ func (vts *VektorSuite) SetupTest() {
 	server := vk.New(
 		vk.UseLogger(logger),
 		vk.UseAppName("vk tester"),
-		vk.UseEnvPrefix("APP_"),
+		vk.UseEnvPrefix("APP"),
```
Comment (arbourd) [inline vk/test/routes_test.go:35]: Purposely reversed this to `APP` for testing purposes. Do we want two tests / servers to ensure both work?

**Review by cohix — APPROVED** [review]

**Pull request merged**